QtCollider/CMakeLists.txt

```diff
@@ -134,8 +144,20 @@ if(QT_COLLIDER_LANG_CLIENT)
     list(APPEND QT_COLLIDER_SRCS ${QT_COLLIDER_DIR}/LanguageClient.cpp)
 endif()
 
+set(QT_COLLIDER_WEBENGINE_HDRS
+  ${QT_COLLIDER_DIR}/widgets/QcWebView.h
+  ${QT_COLLIDER_DIR}/widgets/web_page.hpp
+)
+
+if(SC_USE_WEBENGINE)
+  list(APPEND QT_COLLIDER_HDRS ${QT_COLLIDER_WEBENGINE_HDRS})
+  list(APPEND QT_COLLIDER_SRCS
+    ${QT_COLLIDER_DIR}/widgets/QcWebView.cpp
+    ${QT_COLLIDER_DIR}/widgets/web_page.cpp
+  )
+endif()
```

Comment:
Here's a diff. 

```diff
diff --git a/QtCollider/CMakeLists.txt b/QtCollider/CMakeLists.txt
index f409979fe..f9da30da9 100644
--- a/QtCollider/CMakeLists.txt
+++ b/QtCollider/CMakeLists.txt
@@ -127,2 +127,3 @@ set( QT_COLLIDER_SRCS
   ${QT_COLLIDER_DIR}/widgets/QcLevelIndicator.cpp
+  ${QT_COLLIDER_DIR}/widgets/QcWebView.cpp
   ${QT_COLLIDER_DIR}/widgets/QcWindow.cpp
@@ -135,2 +136,3 @@ set( QT_COLLIDER_SRCS
   ${QT_COLLIDER_DIR}/widgets/QcPenPrinter.cpp
+  ${QT_COLLIDER_DIR}/widgets/web_page.cpp
   ${QT_COLLIDER_DIR}/layouts/layouts.cpp
@@ -153,6 +155,2 @@ if(SC_USE_WEBENGINE)
   list(APPEND QT_COLLIDER_HDRS ${QT_COLLIDER_WEBENGINE_HDRS})
-  list(APPEND QT_COLLIDER_SRCS
-    ${QT_COLLIDER_DIR}/widgets/QcWebView.cpp
-    ${QT_COLLIDER_DIR}/widgets/web_page.cpp
-  )
 endif()
diff --git a/QtCollider/widgets/QcWebView.cpp b/QtCollider/widgets/QcWebView.cpp
index b10061ef6..ffb6ac450 100644
--- a/QtCollider/widgets/QcWebView.cpp
+++ b/QtCollider/widgets/QcWebView.cpp
@@ -21,2 +21,4 @@

+#ifdef SC_USE_WEBENGINE
+
 #include "QcWebView.h"
@@ -323 +325,3 @@ void WebView::onLinkClicked(const QUrl &url, QWebEnginePage::NavigationType type
 } // namespace QtCollider
+
+#endif // SC_USE_WEBENGINE
diff --git a/QtCollider/widgets/web_page.cpp b/QtCollider/widgets/web_page.cpp
index 6f82b2209..4de262b42 100644
--- a/QtCollider/widgets/web_page.cpp
+++ b/QtCollider/widgets/web_page.cpp
@@ -21,2 +21,4 @@

+#ifdef SC_USE_WEBENGINE
+
 #include "web_page.hpp"
@@ -64 +66,2 @@ bool WebPage::acceptNavigationRequest(const QUrl &url, QWebEnginePage::Navigatio

+#endif // SC_USE_WEBENGINE
diff --git a/editors/sc-ide/CMakeLists.txt b/editors/sc-ide/CMakeLists.txt
index 89dac7c91..5f8bf1467 100644
--- a/editors/sc-ide/CMakeLists.txt
+++ b/editors/sc-ide/CMakeLists.txt
@@ -339,2 +339,6 @@ endif()

+if(SC_USE_WEBENGINE)
+    message(STATUS "IDE: Building with QtWebEngine")
+    target_compile_definitions(SuperCollider PUBLIC SC_USE_WEBENGINE)
+endif()

diff --git a/lang/CMakeLists.txt b/lang/CMakeLists.txt
index 05cd992bc..879e0e655 100644
--- a/lang/CMakeLists.txt
+++ b/lang/CMakeLists.txt
@@ -185,3 +185,3 @@ if(SC_QT OR SC_IDE)
   if(SC_USE_WEBENGINE)
-    message(STATUS "Building with QtWebEngine")
+    message(STATUS "sclang: Building with QtWebEngine")
     target_compile_definitions(libsclang PUBLIC SC_USE_WEBENGINE)
```

Reply:
@snappizz Some of these QtCollider headers are added in the SuperCollider (IDE) target, too. With this patch things build cleanly on my machine.

For future reference: the way I figured this out is by looking at the compile commands during building (you can get this with VERBOSE=1 in make or `-v` in Ninja). I paid attention to the use of `-DSC_USE_WEBENGINE` in those commands. Sometimes it was being passed but sometimes it wasn't. In the case that it wasn't, the object being compiled looked like:

`editors/sc-ide/CMakeFiles/SuperCollider.dir/__/__/QtCollider/widgets/moc_web_page.cpp.o`

When building in CMake, whenever you see a path that includes `meow.dir`, it means that that's the working directory for compiling the `meow` target. (At least, that's what I've been able to surmise) So in fact there are two times this file is compiled - once for sclang, and once for IDE. The second time, `-DSC_USE_WEBENGINE` was not being passed. :S